server/index.js

```diff
@@ -66,6 +66,29 @@ app.get('/api/posts/:postId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/posts', (req, res, next) => {
+  const sql = `
+    select "postId",
+           "imageUrl",
+           "summary",
+           "title",
+           "username",
+           "createdAt",
+           "body"
+    from "posts"
+    join "users" using ("userId")
+    order by "postId" desc
+  `;
+  db.query(sql)
+    .then(result => {
+      if (!result.rows) {
+        throw new ClientError(400, 'no posts have been created');
+      }
```

Comment:
This will never actually happen. Even if the array is empty, it's still truthy. You can send an empty list, it's still a 200. You just want to include a conditional render in your list component that shows a message indicating that no posts were returned.

Reply:
Thanks for the feedback, made a a no results page, and am pointing to that conditionally if there are no posts!

![Screen Shot 2021-11-04 at 1 46 21 PM](https://user-images.githubusercontent.com/74282257/140417915-c34d663a-2dd9-4a76-bb2e-b844fc5de707.png)